File: ml/js/chat/geminiChat.js
import { GoogleGenerativeAI } from "@google/generative-ai";

class GeminiChat {
    constructor(apiKey) {
        this.genAI = new GoogleGenerativeAI(apiKey);
        this.model = this.genAI.getGenerativeModel({ model: "gemini-2.0-flash" });
    }

    async sendMessage(history, message, language = 'en') {
        const chat = this.model.startChat({
            history: history,
            generationConfig: {
                maxOutputTokens: 1000,
            },
        });

        const prompt = `[Language: ${language}] ${message}`;
        const result = await chat.sendMessage(prompt);
        return result.response.text();
    }
}

export default GeminiChat;

File: ml/python/chat/gemini_chat.py
import google.generativeai as genai

class GeminiChat:
    def __init__(self, api_key):
        genai.configure(api_key=api_key)
        self.model = genai.GenerativeModel('gemini-2.0-flash')

    def send_message(self, history, message, language='en'):
        chat = self.model.start_chat(history=history)
        prompt = f"[Language: {language}] {message}"
        response = chat.send_message(prompt)
        return response.text
